### ISSUE TITLE
Random failure in aest_sync_SUITE

### ISSUE THREAD

**Issue by uwiger** [aeternity/aeternity#4043]
## Expected Behavior

Passing system test, even in the presence of trivial errors.

## Actual Behavior

Any entry in the crash.log will cause the test to fail, for example:
```
*** User 2022-11-07 11:04:54.485 ***
Node node1's crash logs is not empty:
2022-11-07 11:04:46 =SUPERVISOR REPORT====
     Supervisor: {local,aec_connection_sup}
     Context:    shutdown_error
     Reason:     {shutdown,{gen_server,call,[aec_peers,{get_connection,<<110,24,196,223,45,184,46,170,246,151,230,202,58,123,68,155,139,69,7,137,185,188,110,90,7,215,166,71,48,55,219,109>>}]}}
     Offender:   [{pid,<0.2004.0>},{id,aec_tx_pool_sync},{mfargs,{aec_tx_pool_sync,start_link,[]}},{restart_type,permanent},{shutdown,5000},{child_type,worker}]
```

## Steps to Reproduce the Problem

  1. Happens pretty regularly during CI

## Suggestion
In this particular case, it might be warranted to catch that particular error and return an `{error, not_running}` instead, given that the API already supports sync being stopped, e.g. during maintenance or offline mode.

**Comment by uwiger** [comment]
#3475

**Comment by uwiger** [comment]
Fixed in #4044
